core/sdk-core/src/main/java/software/amazon/awssdk/core/http/NewJsonResponseHandler.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.http;
+
+import static software.amazon.awssdk.utils.Validate.paramNotNull;
+
+import java.util.function.Function;
+import software.amazon.awssdk.annotations.ReviewBeforeRelease;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.core.SdkStandardLogger;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.internal.protocol.json.unmarshall.JsonProtocolUnmarshaller;
+import software.amazon.awssdk.core.protocol.SdkPojo;
+import software.amazon.awssdk.http.SdkHttpFullResponse;
+import software.amazon.awssdk.utils.FunctionalUtils;
+import software.amazon.awssdk.utils.IoUtils;
+
+/**
+ * Default implementation of HttpResponseHandler that handles a successful response from a
+ * service and unmarshalls the result using a JSON unmarshaller.
+ *
+ * @param <T> Indicates the type being unmarshalled by this response handler.
+ */
+@SdkProtectedApi
+@ReviewBeforeRelease("Metadata in base result has been broken. Fix this and deal with AwsResponseHandlerAdapter")
+public final class NewJsonResponseHandler<T extends SdkPojo> implements HttpResponseHandler<T> {
```

Comment:
Name feels little ambiguous. Are we going to replace the current JsonResponseHandler with this class?

Reply:
Yeah will do, meant to replace it.